.github/workflows/preflight-release.yml

```diff
@@ -77,7 +76,7 @@ jobs:
           sed -i "s|quay.io/opdev/preflight:[0-9]*\.[0-9]*\.[0-9]*|quay.io/opdev/preflight:${PREFLIGHT_LATEST}|" roles/preflight/defaults/main.yml
           sed -i "s|quay.io/opdev/preflight:[0-9]*\.[0-9]*\.[0-9]*|quay.io/opdev/preflight:${PREFLIGHT_LATEST}|" roles/preflight/README.md
           git add roles/preflight/defaults/main.yml roles/preflight/README.md
-          git commit --allow-empty -m "Update Preflight to release $PREFLIGHT_LATEST"
+          git commit -s -m "Update Preflight to release $PREFLIGHT_LATEST"
```

Comment:
Github ate the second part of my reply, but: If you remove `--allow-empty` and the commit is actually empty, this will fail. Also adding `-s` means you have to configure GIT_AUTHOR and GIT_EMAIL in the GHA environment as it adds a `--Signed-off-by` note

Reply:
To remove `--allow-empty` is fine, it was only here for tests.
Concerning `-s`, I tried to fix the workflows strangely pending in the `expected` status in the [bot-opened PR](https://github.com/redhatci/ansible-collection-redhatci-ocp/pull/339). Thank you for the advice, let me add the config for the PR sign-off.

Reply:
ok, so now the commit is signed: https://github.com/redhatci/ansible-collection-redhatci-ocp/pull/343
but it didn't solve the issue with the pending actions

Reply:
@tkrishtop sorry, but I'm not sure how is it going to be possible to merge the commits created by this process :sweat_smile: , since they are not  [signed](https://github.com/redhatci/ansible-collection-redhatci-ocp/blob/main/docs/dev.md#sign-commits)

Merges require signed commits for verification. Btw, `--signed-off-by` is effectively just a comment in the commit that provides acknowledgment but is not an actual verification of ownership.

For example, this commit 82d6c32f998663738ab73ae482883bc0b3337a39 is signed, verified with:
```Shell
❯ git verify-commit 82d6c32f998663738ab73ae482883bc0b3337a39; echo $?
gpg: Signature made Mon 10 Jun 2024 04:49:44 AM CDT
gpg:                using EDDSA key 227039C11A644C04D3672F01A1F56AFB08EA13C6
gpg:                issuer "nsilla@redhat.com"
gpg: Good signature from "Nacho Silla <nsilla@redhat.com>" [full]
0
```

But this other one is not :open_mouth: 8270c0eb0fadf3cddb237be184b94c31a6a62948

```Shell
❯ git verify-commit 8270c0eb0fadf3cddb237be184b94c31a6a62948; echo $?
1
```

Reply:
Hi @tonyskapunk here is the commit you're [pointing to](https://github.com/redhatci/ansible-collection-redhatci-ocp/pull/346/commits/8270c0eb0fadf3cddb237be184b94c31a6a62948): 
![image](https://github.com/redhatci/ansible-collection-redhatci-ocp/assets/26544656/daaf20ae-a808-44da-8063-43048566e8d3)

It's signed by the email but we have no GPG, that's true. Not sure if that maybe a reason of the GH actions stuck in the 'executed' stage forever in the [PR opened by a bot](https://github.com/redhatci/ansible-collection-redhatci-ocp/pull/346).

Reply:
@tonyskapunk @thekad do you think it's a good idea to add GPG key for the bot through the secrets? 
```
- name: Import GPG key
  env:
    GPG_PRIVATE_KEY: ${{ secrets.GPG_PRIVATE_KEY }}
    GPG_PASSPHRASE: ${{ secrets.GPG_PASSPHRASE }}
  run: |
    echo "$GPG_PRIVATE_KEY" | gpg --batch --import
    echo "$GPG_PASSPHRASE" | gpg --batch --yes --passphrase-fd 0 --pinentry-mode loopback

- name: Configure git for GPG
  run: |
    git config --global user.signingkey $(gpg --list-secret-keys --keyid-format LONG | grep sec | awk '{print $2}' | cut -d '/' -f 2)
    git config --global gpg.program gpg
    git config --global commit.gpgSign true
    git config --global user.name "github-actions[bot]"
    git config --global user.email "github-actions[bot]@users.noreply.github.com"

- name: Create new branch with updated Preflight version
  if: steps.compare_versions.outputs.new_release == 'true' && steps.check_branch.outputs.branch_exists == 'false'
  env:
    PREFLIGHT_LATEST: ${{ steps.check_release.outputs.preflight_latest }}
  run: |
    # Pull the latest changes from the main branch
    git fetch origin main
    git checkout main
    git pull origin main
    # Create a new branch from the main branch
    git checkout -b preflight-release-$PREFLIGHT_LATEST
    # Update preflight version
    sed -i "s|quay.io/opdev/preflight:[0-9]*\.[0-9]*\.[0-9]*|quay.io/opdev/preflight:${PREFLIGHT_LATEST}|" roles/preflight/defaults/main.yml
    sed -i "s|quay.io/opdev/preflight:[0-9]*\.[0-9]*\.[0-9]*|quay.io/opdev/preflight:${PREFLIGHT_LATEST}|" roles/preflight/README.md
    git add roles/preflight/defaults/main.yml roles/preflight/README.md
    git commit -S -m "Update Preflight to release $PREFLIGHT_LATEST"
    git push --set-upstream origin preflight-release-$PREFLIGHT_LATEST
```

Reply:
@nsilla we're still discussing here, one more issue to address

Reply:
Using gpg keys through secrets is the way to go, let's add them and see if there won't be any issue when merging, as it won't be possible to add the pub key into the user's account